comp/core/log/mock.go

```diff
@@ -0,0 +1,51 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package log
+
+import (
+	"context"
+	"strings"
+	"testing"
+
+	"github.com/cihub/seelog"
+	"go.uber.org/fx"
+
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+// tbWriter is an implementation of io.Writer that sends lines to
+// testing.TB#Log.
+type tbWriter struct {
+	t testing.TB
+}
+
+// Write implements Writer#Write.
+func (tbw *tbWriter) Write(p []byte) (n int, err error) {
+	// this assumes that seelog always writes one log entry in one Write call
+	msg := strings.TrimSpace(string(p))
```

Comment:
Why trim spaces ? Wouldn't this be a different behavior than the real logger ?

Reply:
Mostly to trim the trailing newline.  We're not trying to match behavior here, but to provide useful debugging information, so I think it's not likely to lead to any confusion.

Reply:
(I'll push a commit to use strings.TrimSuffix)